src/interpreter/plugin/DateTimePlugin.ts

```diff
@@ -596,76 +596,41 @@ export class DateTimePlugin extends FunctionPlugin {
   private networkdayscore(start: number, end: number, weekend: InternalNoErrorCellValue, holidays?: SimpleRangeValue): InternalScalarValue {
     start = Math.trunc(start)
     end = Math.trunc(end)
+    let multiplier = 1
     if(start>end) {
       [start, end] = [end, start]
+      multiplier = -1
     }
-    if(typeof weekend !== 'number' && typeof weekend !== 'string') {
-      return new CellError(ErrorType.VALUE, ErrorMessage.WrongType)
-    }
-
-    let weekendPattern: Maybe<string>
 
-    if(typeof weekend === 'string') {
-      if(weekend.length !== 7 || !/^(0|1)*$/.test(weekend)) {
-        return new CellError(ErrorType.NUM, ErrorMessage.WeekendString)
-      } else {
-        weekendPattern = weekend
-      }
-    } else {
-      weekendPattern = workdayPatterns.get(weekend)
-      if(weekendPattern === undefined) {
-        return new CellError(ErrorType.NUM, ErrorMessage.BadMode)
-      }
+    const weekendPattern = computeWeekendPattern(weekend)
+    if(weekendPattern instanceof CellError) {
+      return weekendPattern
     }
 
-    const uniqueHolidays = (holidays !== undefined) ? simpleRangeToUniqueNumbers(holidays) : []
-    if(uniqueHolidays instanceof CellError) {
-      return uniqueHolidays
+    const filteredHolidays = this.simpleRangeToFilteredHolidays(holidays, weekendPattern)
+    if(filteredHolidays instanceof CellError) {
+      return filteredHolidays
     }
-
-    const filteredHolidays = uniqueHolidays.filter((arg) => {
-      const val = this.interpreter.dateHelper.relativeNumberToAbsoluteNumber(arg)
-      const i = (val-1)%7
-      return (weekendPattern!.charAt(i) === '0')
-    })
-
-    return this.countWorkdays(start, end, weekendPattern, filteredHolidays)
+    return multiplier * this.countWorkdays(start, end, weekendPattern, filteredHolidays)
   }
 
   private workdaycore(start: number, delta: number, weekend: InternalNoErrorCellValue, holidays?: SimpleRangeValue): InternalScalarValue {
     start = Math.trunc(start)
     delta = Math.trunc(delta)
 
-    if(typeof weekend !== 'number' && typeof weekend !== 'string') {
-      return new CellError(ErrorType.VALUE, ErrorMessage.WrongType)
+    if(weekend === '1111111') {
```

Comment:
Do you need it here? IIUC, you'll check it in the next lines anyway.

Reply:
thanks for spotting an error. NETWORKDAYS accepts '1111111', while WORKDAY does not

Reply:
ok, actually NETWORKDAYS does not accept '1111111' even though it makes sense there...

Reply:
Yeah, 1111111 is quite an awkward setting :)